lib/scope_file_access_flat.rb

```diff
@@ -30,7 +30,7 @@ def binary?(system_file)
     path = system_file.scope.file_path(system_file)
     return false if File.zero?(path)
 
-    Machinery.content_is_binary?(File.read(path))
+    Machinery.content_is_binary?(File.read(path).slice(0, 1024))
```

Comment:
Instead of reading the whole file into memory and passing on the small slice you can also just read the amount of data you want (e.g. `File.read(path, 1024)`).